src/XCCDF_POLICY/xccdf_policy.c

```diff
@@ -981,6 +979,52 @@ _xccdf_policy_rule_evaluate(struct xccdf_policy * policy, const struct xccdf_rul
 		return _xccdf_policy_report_rule_result(policy, result, rule, NULL, XCCDF_RESULT_NOT_SELECTED, NULL);
 	}
 
+	bool all_requires_met = true;
```

Comment:
the impl for this shouldn't be here, this is slow and unnecessary. we need to do this at resolve stage.